src/main/java/com/alipay/oceanbase/rpc/location/model/partition/ObHashPartDesc.java

```diff
@@ -98,39 +101,47 @@ public void prepare() {
      * Get part ids.
      */
     @Override
-    public List<Long> getPartIds(Object[] start, boolean startInclusive, Object[] end,
+    public List<Long> getPartIds(Object startRowObj, boolean startInclusive, Object endRowObj,
                                  boolean endInclusive) {
         // close set
         try {
+            // verify the type of parameters and convert to Row
+            if (!(startRowObj instanceof Row) || !(endRowObj instanceof Row)) {
+                throw new ObTableException("invalid format of rowObj: " + startRowObj + ", "
+                                           + endRowObj);
+            }
+            Row startRow = (Row) startRowObj, endRow = (Row) endRowObj;
             // pre-check start and end
             // should remove after remove addRowkeyElement
-            if (start.length != end.length) {
+            if (startRow.size() != endRow.size()) {
                 throw new IllegalArgumentException("length of start key and end key is not equal");
             }
 
             // check whether partition key is Min or Max, should refactor after remove addRowkeyElement
-            for (ObPair<ObColumn, List<Integer>> pair : orderedPartRefColumnRowKeyRelations) {
-                for (int refIdx : pair.getRight()) {
-                    if (start.length <= refIdx) {
-                        throw new IllegalArgumentException("rowkey length is " + start.length
+            for (ObColumn curObcolumn : partColumns) {
+                for (int refIdx = 0; refIdx < curObcolumn.getRefColumnNames().size(); ++refIdx) {
+                    String curObRefColumnName = curObcolumn.getRefColumnNames().get(refIdx);
+                    if (startRow.size() <= refIdx) {
+                        throw new IllegalArgumentException("rowkey length is " + startRow.size()
                                                            + ", which is shortest than " + refIdx);
                     }
-                    if (start[refIdx] instanceof ObObj
-                        && (((ObObj) start[refIdx]).isMinObj() || ((ObObj) start[refIdx])
-                            .isMaxObj())) {
+                    if (startRow.get(curObRefColumnName) instanceof ObObj
```

Comment:
what happens if the startRow do not contain the `curObRefColumnName`？

Reply:
Already changed. Same in ObKeyPartDesc.